capabilityreceiver/pkg/capabilityhandler/includedfields.go

```diff
@@ -88,3 +101,170 @@ func checkField(capInt map[string]interface{}, fieldNames []string) bool {
+		{"rest", "interaction", "modifierExtension", "http://hl7.org/fhir/StructureDefinition/capabilitystatement-prohibited", "capabilitystatement-prohibited"},
+	}
+
+	// Get name of extension and create extensionObj with extension name, if the extension exists, and if it is an extension
+	for _, extensionPath := range extensionList {
+		extensionName := extensionPath[len(extensionPath)-1]
+		extensionURL := extensionPath[len(extensionPath)-2]
+		//Check if includedFields already contains this extension
+		index := includedFieldsContains(includedFields, extensionName)
+		if index != -1 {
+			//If includedFields contains extension but Exists is false, check next possible location to see if it exists
+			if !includedFields[index].Exists {
+				includedFields[index].Exists = checkExtension(capInt, extensionPath, extensionURL)
+			} else {
+				continue
+			}
+		} else {
+			extensionObj := endpointmanager.IncludedField{
+				Field:     extensionName,
+				Exists:    checkExtension(capInt, extensionPath, extensionURL),
+				Extension: true,
+			}
+			includedFields = append(includedFields, extensionObj)
+		}
+	}
+
+	return includedFields
+}
+
+// Checks whether the extension is populated in the capability statement given a path of fieldNames
+func checkExtension(capInt map[string]interface{}, fieldNames []string, url string) bool {
+	for index, name := range fieldNames {
+		if capInt[name] == nil {
+			return false
+		}
+
+		field := capInt[name]
+		// Check if at an extension field in fieldNames
+		if index == len(fieldNames)-3 {
+			extensionArr := field.([]interface{})
+			return checkExtensionURL(extensionArr, url)
+		} else if arrContains(arrayFields, name) {
+			fieldArr := field.([]interface{})
+			nextIndex := index + 1
+			length := len(fieldNames)
+			return checkArrFieldExtension(fieldNames[nextIndex:length], fieldArr, url)
+		} else {
+			capInt = field.(map[string]interface{})
+		}
+	}
+
+	return false
+}
+
+// Given an array of interface objects, loops through each object to check whether the extension is populated following the path of fieldNames
+func checkArrFieldExtension(fieldNames []string, fieldArr []interface{}, url string) bool {
+	var found bool
+	// Loop through the array of interface objects
+	for _, obj := range fieldArr {
+		// For each object in interface array, get desired field using name in fieldNames
+		name := fieldNames[0]
+		objMap := obj.(map[string]interface{})
+		extensionField := objMap[name]
+		if extensionField == nil {
+			// If the desired field does not exist in that object, continue to the next object within the array of interface objects
+			continue
+		} else if len(fieldNames)-3 != 0 && arrContains(arrayFields, name) {
+			// If the desired field is not extension or modifierExtension and it is also an array of interface objects, call checkArrFieldExtension with this new array
+			fieldArr := extensionField.([]interface{})
+			length := len(fieldNames)
```

Comment:
length could be defined outside the if chain here and then also used both places you do len(fieldNames)-3